### DISCUSSION TITLE
Format difference between string and Form.useWatcher()

### DISCUSSION THREAD

**Discussion by laux98** [ant-design/ant-design#43540] [Q&A] [answered]
Hello everyone, I'm stuck since few hours with Form.useInstance() and I wouldn't say no to a little helping hand.

So, this is my form : 

```tsx
const [myForm] = Form.useForm();
const watcherData = Form.useWatch("data", formPortSerie);

return 
  <Form wrapperCol={{ xs: 14 }} labelCol={{ xs: 10 }} name="basic" form={myForm} colon={false}>
    <Form.Item name="data" label="Some data :">
      <Input />
    </Form.Item>
  </Form>
```

With this little form, I want to get a command for send it to Serial Port.

In my field data, I type this string : 

```
\u001bp1\n\r
```

The problem is that at the end of the form I don't have the same result stored in the watcher as if I display the variable directly with console.log() : 

In the code : 

```js 
console.log("watcher : " , watcherData);
console.log("\u001bp1\n\r");
```

The result in console :

```
watcher:  \u001bp1\n\r
p1
```

So my goal is to get the string: p1 in my watcher.

Thanks for reading and thanks in advance to anyone who tries to help me.

**Comment by linxianxi** [comment, marked as answer]
try https://www.npmjs.com/package/unraw
`unraw("\u001bp1\n\r")`

**Reply by laux98** [reply]
Thank you very much, it works perfectly!

**Comment by zombieJ** [comment]
Your typing in the input will get `\\u001bp1\\n\\r`. Web will auto convert this. Which is not `\u001bp1\n\r`

**Reply by laux98** [reply]
Thank you very much for your explanations
